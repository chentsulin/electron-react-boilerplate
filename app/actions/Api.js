// @flow
import type { counterStateType } from '../reducers/Api';

type actionType = {
  +type: string
};

export const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
export const DECREMENT_COUNTER = 'DECREMENT_COUNTER';

export function increment() {
  return {
    type: INCREMENT_COUNTER
  };
}

export function decrement() {
  return {
    type: DECREMENT_COUNTER
  };
}

export function incrementIfOdd() {
  return (dispatch: (action: actionType) => void, getState: () => counterStateType) => {
    const { counter } = getState();

    if (counter % 2 === 0) {
      return;
    }

    dispatch(increment());
  };
}

export function incrementAsync(delay: number = 1000) {
  return (dispatch: (action: actionType) => void) => {
    setTimeout(() => {
      dispatch(increment());
    }, delay);
  };
}

// nicks new stuff
export function logResult(result) {
  console.log(result);
}

export function logError(error) {
  console.log('Looks like there was a problem: \n', error);
}

export function validateResponse(response) {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response;
}

export function readResponseAsJSON(response) {
  return response.json();
}

export function fetchJSON(pathToResource) {
  fetch(pathToResource) // 1
  .then(validateResponse) // 2
  .then(readResponseAsJSON) // 3
  .then(logResult) // 4
  .catch(logError);
}
