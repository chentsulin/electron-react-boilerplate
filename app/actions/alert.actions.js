import { alertConstants } from '../constants';

export const alertActions = {
  success,
  error,
  clear
};

export default alertActions;

function success(message) {
  return { type: alertConstants.SUCCESS, message };
}

function error(message) {
  return { type: alertConstants.ERROR, message };
}

function clear() {
  return { type: alertConstants.CLEAR };
}
