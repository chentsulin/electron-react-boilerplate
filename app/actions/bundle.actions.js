import sort from 'fast-sort';
import traverse from 'traverse';
import { bundleConstants } from '../constants/bundle.constants';
import { bundleService } from '../services/bundle.service';

export const bundleActions = {
  mockFetchAll,
  fetchAll,
  delete: removeBundle,
  downloadResources,
  requestSaveBundleTo,
  toggleModePauseResume,
  toggleSelectBundle
};

export default bundleActions;

export function mockFetchAll() {
  return dispatch => {
    dispatch(request());
    return new Promise(resolve => {
      const mockBundles = getMockBundles();
      resolve(mockBundles);
    }).then(bundles => dispatch(success(bundles)), error => dispatch(failure(error)));
  };

  function request() {
    return { type: bundleConstants.FETCH_REQUEST };
  }
  function success(bundles) {
    return { type: bundleConstants.FETCH_SUCCESS, bundles };
  }
  function failure(error) {
    return { type: bundleConstants.FETCH_FAILURE, error };
  }
}

export function fetchAll() {
  return dispatch => {
    dispatch(request());

    return bundleService
      .fetchAll()
      .then(bundles => dispatch(success(bundles)), error => dispatch(failure(error)));
  };

  function request() {
    return { type: bundleConstants.FETCH_REQUEST };
  }
  function success(bundles) {
    return { type: bundleConstants.FETCH_SUCCESS, bundles };
  }
  function failure(error) {
    return { type: bundleConstants.FETCH_FAILURE, error };
  }
}

export function downloadResources(id) {
  return async dispatch => {
    try {
      const manifestResourcePaths = await bundleService.getManifestResourcePaths(id);
      manifestResourcePaths.unshift('metadata.xml');
      await bundleService.downloadResources(id);
      dispatch(request(id, manifestResourcePaths));
    } catch (error) {
      dispatch(failure(id, error));
    }
  };
  function request(_id, manifestResourcePaths) {
    return { type: bundleConstants.DOWNLOAD_RESOURCES_REQUEST, id: _id, manifestResourcePaths };
  }
  function failure(_id, error) {
    return { type: bundleConstants.DOWNLOAD_RESOURCES_FAILURE, id, error };
  }
}


function removeBundle(id) {
  return dispatch => {
    dispatch(request(id));

    bundleService
      .delete(id)
      .then(() => {
        dispatch(success(id));
        return true;
      })
      .catch(error => {
        dispatch(failure(id, error));
        return true;
      });
  };

  function request(_id) {
    return { type: bundleConstants.DELETE_REQUEST, id: _id };
  }
  function success(_id) {
    return { type: bundleConstants.DELETE_SUCCESS, id: _id };
  }
  function failure(_id, error) {
    return { type: bundleConstants.DELETE_FAILURE, id: _id, error };
  }
}

export function requestSaveBundleTo(id, selectedFolder) {
  return async dispatch => {
    const bundleInfo = await bundleService.fetchById(id);
    const bundleBytesToSave = traverse(bundleInfo.store.file_info).reduce(addByteSize, 0);
    const resourcePaths = await bundleService.getResourcePaths(id);
    resourcePaths.unshift('metadata.xml');
    const resourcePathsProgress = resourcePaths.reduce((acc, resourcePath) => {
      acc[resourcePath] = 0;
      return acc;
    }, {});
    let bundleBytesSaved = 0;
    dispatch(request(id, selectedFolder, bundleBytesToSave, resourcePaths));
    resourcePaths.forEach(async resourcePath => {
      try {
        const downloadItem = await bundleService.requestSaveResourceTo(
          selectedFolder,
          id,
          resourcePath,
          (resourceTotalBytesSaved, resourceProgress) => {
            const originalResourceBytesTransferred = resourcePathsProgress[resourcePath];
            resourcePathsProgress[resourcePath] = resourceTotalBytesSaved;
            const bytesDiff = resourceTotalBytesSaved - originalResourceBytesTransferred;
            bundleBytesSaved += bytesDiff;
            if (resourceProgress && resourceProgress % 100 === 0) {
              dispatch(updated(
                id,
                resourcePath,
                resourceTotalBytesSaved,
                bundleBytesSaved,
                bundleBytesToSave
              ));
            }
          }
        );
        return downloadItem;
      } catch (error) {
        dispatch(failure(id, error));
      }
    });
  };

  function addByteSize(accBytes, fileInfoNode) {
    if (fileInfoNode.is_dir || this.isRoot || fileInfoNode.size === undefined) {
      return accBytes;
    }
    return accBytes + fileInfoNode.size;
  }

  function request(_id, _folderName, bundleBytesToSave, resourcePaths) {
    return {
      type: bundleConstants.SAVETO_REQUEST,
      id: _id,
      folderName: _folderName,
      bundleBytesToSave,
      resourcePaths
    };
  }

  function updated(
    _id,
    resourcePath,
    resourceTotalBytesSaved,
    bundleBytesSaved,
    bundleBytesToSave
  ) {
    return {
      type: bundleConstants.SAVETO_UPDATED,
      id,
      resourcePath,
      resourceTotalBytesSaved,
      bundleBytesSaved,
      bundleBytesToSave
    };
  }
  function failure(_id, error) {
    return { type: bundleConstants.SAVETO_FAILURE, id: _id, error };
  }
}

export function toggleModePauseResume(id) {
  return { type: bundleConstants.TOGGLE_MODE_PAUSE_RESUME, id };
}

export function toggleSelectBundle(id) {
  return { type: bundleConstants.TOGGLE_SELECT, id };
}

function getMockBundles() {
  const bundles = [
    {
      id: 'bundle01',
      name: 'Test Bundle #1',
      revision: 3,
      task: 'UPLOAD',
      status: 'COMPLETED'
    },
    {
      id: 'bundle02',
      name: 'Another Bundle',
      revision: 3,
      task: 'UPLOAD',
      status: 'IN_PROGRESS',
      progress: 63,
      mode: 'PAUSED'
    },
    {
      id: 'bundle03',
      name: 'Audio Bundle',
      revision: 52,
      task: 'DOWNLOAD',
      status: 'IN_PROGRESS',
      progress: 12,
      mode: 'RUNNING'
    },
    {
      id: 'bundle04',
      name: 'Unfinished Bundle',
      task: 'UPLOAD',
      status: 'DRAFT'
    },
    {
      id: 'bundle05',
      name: 'Unfinished Video Bundle',
      task: 'UPLOAD',
      status: 'DRAFT'
    },
    {
      id: 'bundle06',
      name: 'DBL Bundle',
      task: 'DOWNLOAD',
      status: 'NOT_STARTED'
    },
    {
      id: 'bundle07',
      name: 'DBL Bundle 3',
      task: 'DOWNLOAD',
      status: 'NOT_STARTED'
    },
    {
      id: 'bundle08',
      name: 'Audio Bundle #2',
      revision: 40,
      task: 'DOWNLOAD',
      status: 'COMPLETED'
    },
    {
      id: 'bundle09',
      name: 'Audio Bundle #3',
      revision: 5,
      task: 'SAVETO',
      status: 'IN_PROGRESS',
      progress: 0,
    },
    {
      id: 'bundle10',
      name: 'Audio Bundle #4',
      revision: 4,
      task: 'SAVETO',
      status: 'IN_PROGRESS',
      progress: 66,
    },
    {
      id: 'bundle11',
      name: 'Audio Bundle #5',
      revision: 5,
      task: 'SAVETO',
      status: 'COMPLETED',
      progress: 100,
    }
  ];
  const taskOrder = ['UPLOAD', 'DOWNLOAD', 'SAVETO'];
  const statusOrder = ['IN_PROGRESS', 'DRAFT', 'COMPLETED', 'NOT_STARTED'];
  const sortedBundles = sort(bundles).asc([
    b => statusOrder.indexOf(b.status),
    b => taskOrder.indexOf(b.task),
    b => b.name
  ]);
  return sortedBundles;
}
