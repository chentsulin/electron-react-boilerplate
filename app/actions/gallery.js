/**
 * Test List actions
 */
export const SAVE_SCROLL_POSITION = 'redux/Gallery/SAVE_SCROLL_POSITION';
export const MARK_ITEM = 'redux/Gallery/MARK_ITEM';
export const LOAD_ITEMS = 'redux/Gallery/LOAD_ITEMS';
export const LOAD_ITEMS_SUCCESS = 'redux/Gallery/LOAD_ITEMS_SUCCESS';
export const LOAD_ITEMS_ERROR = 'redux/Gallery/LOAD_ITEMS_ERROR';
export const LOAD_ITEM_SUCCESS = 'redux/Gallery/LOAD_ITEM_SUCCESS';
export const LOAD_ITEM_ERROR = 'redux/Gallery/LOAD_ITEM_ERROR';

/**
 * Mark an item.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param {string} id - The item Id.
 * @param {boolean} marked - The new item marked state.
 * @return {object} - An action object with a type of MARK_ITEM
 */
export function markItem(moduleId, galleryId, searchQuery, id, marked) {
  return {
    type: MARK_ITEM,
    payload: {
      id,
      marked,
    },
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Saves scroll position for later.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param {number} scrollPosition - The scroll position.
 * @param {number} oldWidth - The width when the scroll position was saved.
 * @return {object} - An action object with a type of SAVE_SCROLL_POSITION.
 */
export function saveScrollPosition(moduleId, galleryId, searchQuery, scrollPosition, oldWidth = 0) {
  return {
    type: SAVE_SCROLL_POSITION,
    payload: {
      scrollPosition,
      oldWidth,
    },
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Load the items, this action starts the request saga.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @return {object} - An action object with a type of LOAD_ITEMS.
 */
export function loadItems(moduleId, galleryId, searchQuery) {
  return {
    type: LOAD_ITEMS,
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Dispatched when the items are loaded by the request saga.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param {boolean} hasNextPage - Flag indicating galelry has more pages
 * @return {object} - An action object with a type of LOAD_ITEMS_SUCCESS.
 */
export function itemsLoaded(moduleId, galleryId, searchQuery, hasNextPage) {
  return {
    type: LOAD_ITEMS_SUCCESS,
    payload: {
      hasNextPage,
    },
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Dispatched when loading the items fails.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param  {object} error - The error.
 * @return {object} - An action object with a type of LOAD_ITEMS_ERROR passing the error.
 */
export function itemsLoadingError(moduleId, galleryId, searchQuery, error) {
  return {
    type: LOAD_ITEMS_ERROR,
    payload: error,
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Dispatched when an item is loaded by the request saga.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param  {*} item - The item.
 * @return {object} - An action object with a type of LOAD_ITEM_SUCCESS passing the item.
 */
export function itemLoaded(moduleId, galleryId, searchQuery, item) {
  return {
    type: LOAD_ITEM_SUCCESS,
    payload: item,
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}

/**
 * Dispatched when loading the items fails.
 *
 * @param {string | number} moduleId - The module Id.
 * @param {string | number} galleryId - The gallery Id.
 * @param {string} searchQuery - The search query text.
 * @param  {*} item - The item.
 * @param  {object} error - The error.
 * @return {object} - An action object with a type of LOAD_ITEM_ERROR passing the error.
 */
export function itemLoadingError(moduleId, galleryId, searchQuery, item, error) {
  return {
    type: LOAD_ITEM_ERROR,
    payload: {
      item,
      error,
    },
    meta: {
      moduleId,
      galleryId,
      searchQuery,
    },
  };
}
