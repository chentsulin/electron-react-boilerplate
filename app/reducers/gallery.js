/*
 * Test List reducer
 */
import { fromJS } from 'immutable';

import modules from '../modules';
import { SET_GALLERY } from '../actions/global';
import {
  SAVE_SCROLL_POSITION,
  MARK_ITEM,
  LOAD_ITEMS,
  LOAD_ITEMS_SUCCESS,
  LOAD_ITEMS_ERROR,
  LOAD_ITEM_SUCCESS,
  LOAD_ITEM_ERROR,
} from '../actions/gallery';

const initialGalleryState = {
  loading: false,
  error: false,
  hasNextPage: true,
  pageNumber: 0,
  items: {},
  itemsList: [],
  scrollPosition: -1,
  oldWidth: 0,
};

const initialStateBuilder = {}; // ['module/galleryId'] => { gallery state }

// Load all of our modules
modules.keySeq().forEach(key => {
  initialStateBuilder[key] = { ...initialGalleryState };
});

// The initial state of the App
export const initialState = fromJS(initialStateBuilder);

// fuynction for making path
export const makePath = meta => {
  const { moduleId = undefined, galleryId = undefined, searchQuery = undefined } = meta || {};
  let path = moduleId;

  if (galleryId) {
    path = `/gallery/${path}/${galleryId}`;
  }

  if (searchQuery) {
    path = `/search/${path}/${searchQuery}`;
  }

  return path;
};

function galleryReducer(state = initialState, action) {
  const { type, payload, meta } = action;
  let path = makePath(meta);

  switch (type) {
    case SET_GALLERY: {
      path = makePath(payload);

      if (!state.has(path)) {
        return state.set(path, fromJS({ ...initialGalleryState }));
      }

      return state;
    }
    case SAVE_SCROLL_POSITION: {
      const { scrollPosition, oldWidth } = payload;
      return state.setIn([path, 'scrollPosition'], scrollPosition).setIn([path, 'oldWidth'], oldWidth);
    }
    case MARK_ITEM: {
      const { id, marked } = payload;
      return state.setIn([path, 'items', `${id}`, 'marked'], marked);
    }
    case LOAD_ITEMS:
      return state
        .setIn([path, 'loading'], true)
        .setIn([path, 'error'], false)
        .setIn([path, 'pageNumber'], state.getIn([path, 'pageNumber']) + 1);
    case LOAD_ITEMS_SUCCESS: {
      return state.setIn([path, 'hasNextPage'], payload.hasNextPage).setIn([path, 'loading'], false);
    }
    case LOAD_ITEMS_ERROR:
      return state.setIn([path, 'error'], payload).setIn([path, 'loading'], false);
    case LOAD_ITEM_SUCCESS: {
      const { id } = payload;
      return state
        .setIn([path, 'items', id], fromJS(payload))
        .updateIn([path, 'itemsList'], itemsList => itemsList.push(id));
    }
    case LOAD_ITEM_ERROR: {
      const { item, error } = payload;
      if (process.env.NODE_ENV !== 'test') {
        console.warn(`Cannot load image ${item.image} in gallery ${path}`, error); // eslint-disable-line no-console
      }
      return state;
    }
    default:
      return state;
  }
}

export default galleryReducer;
