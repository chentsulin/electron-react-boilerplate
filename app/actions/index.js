import * as globalActions from './global';
import * as galleryActions from './gallery';
import * as languageActions from './language';

const actions = {
  ...globalActions,
  ...galleryActions,
  ...languageActions,
};

export default actions;
