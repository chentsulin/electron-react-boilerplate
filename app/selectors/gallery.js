/**
 * The Test Lsit state selectors
 */

import { createSelector } from 'reselect';

import { initialState, makePath } from '../reducers/gallery';

const selectGallery = state => state.get('gallery', initialState);

const makeSelectLoading = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'loading']),
  );

const makeSelectError = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'error']),
  );

const makeSelectPageNumber = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'pageNumber']),
  );

const makeSelectHasNextPage = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'hasNextPage']),
  );

const makeSelectScrollPosition = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'scrollPosition']),
  );

const makeSelectOldWidth = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'oldWidth']),
  );

const makeSelectItemsList = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'itemsList']).toJS(),
  );

const makeSelectItems = (moduleId, galleryId, searchQuery) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'items']).toJS(),
  );

const makeSelectItem = (moduleId, galleryId, searchQuery, id) =>
  createSelector(
    selectGallery,
    galleryState => galleryState.getIn([makePath({ moduleId, galleryId, searchQuery }), 'items', `${id}`]).toJS(),
  );

export {
  selectGallery,
  makeSelectPageNumber,
  makeSelectLoading,
  makeSelectError,
  makeSelectHasNextPage,
  makeSelectItem,
  makeSelectItemsList,
  makeSelectItems,
  makeSelectScrollPosition,
  makeSelectOldWidth,
};
