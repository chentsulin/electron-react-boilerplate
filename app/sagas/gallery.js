/**
 * Gets the next batch of items
 */
import { put, select, takeLatest, all, call } from 'redux-saga/effects';

import { loadImageAsync } from '../imageLoader';
import { LOAD_ITEMS, itemsLoaded, itemsLoadingError, itemLoaded, itemLoadingError } from '../actions/gallery';
import { makeSelectPageNumber } from '../selectors/gallery';
import {
  makeSelectCurrentModuleId,
  makeSelectCurrentGalleryId,
  makeSelectModule,
  makeSelectCurrentSearchQuery,
} from '../selectors/global';

export function* getItem(moduleId, galleryId, searchQuery, item) {
  try {
    // Load item
    const img = yield call(loadImageAsync, item.image);

    // Finish
    yield put(
      itemLoaded(moduleId, galleryId, searchQuery, {
        ...item,
        width: img.width,
        height: img.height,
      }),
    );
  } catch (error) {
    console.error('Failed to load item', error); // eslint-disable-line no-console
    yield put(itemLoadingError(moduleId, galleryId, searchQuery, error));
  }
}

/**
 * Items request/response handler
 */
export function* getItems() {
  const moduleId = yield select(makeSelectCurrentModuleId());
  const galleryId = yield select(makeSelectCurrentGalleryId());
  const searchQuery = yield select(makeSelectCurrentSearchQuery());
  const currentModule = yield select(makeSelectModule(moduleId));
  const { module } = currentModule;

  try {
    // Select page number from store
    const page = yield select(makeSelectPageNumber(moduleId, galleryId, searchQuery));

    // Get urls and galelry info
    let result = {};
    if (searchQuery) {
      result = yield call(module.searchGalleriesAsync.bind(module), searchQuery, page);
    } else if (galleryId) {
      result = yield call(module.getGalleryAsync.bind(module), galleryId, page);
    } else {
      result = yield call(module.getGalleriesAsync.bind(module), page);
    }

    // Load all
    yield all(result.items.map(item => call(getItem, moduleId, galleryId, searchQuery, item)));

    // Finish
    yield put(itemsLoaded(moduleId, galleryId, searchQuery, result.hasNextPage));
  } catch (error) {
    console.error('Failed to load items', error); // eslint-disable-line no-console
    yield put(itemsLoadingError(moduleId, galleryId, searchQuery, error));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* gallerySaga() {
  // Watches for LOAD_ITEMS actions and calls getRepos when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(LOAD_ITEMS, getItems);
}
