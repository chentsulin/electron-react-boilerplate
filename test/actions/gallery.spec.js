import * as actions from '../../app/actions/gallery';

describe('actions', () => {
  describe('gallery', () => {
    it('markItem should create MARK_ITEM action', () => {
      expect(actions.markItem(1, 2, 'search', 3, true)).toMatchSnapshot();
    });

    it('saveScrollPosition should create SAVE_SCROLL_POSITION action', () => {
      expect(actions.saveScrollPosition(1, 2, 'search', 3, 4)).toMatchSnapshot();
    });

    it('loadItems should create LOAD_ITEMS action', () => {
      expect(actions.loadItems(1, 2, 'search')).toMatchSnapshot();
    });

    it('itemsLoaded should create LOAD_ITEMS_SUCCESS action', () => {
      expect(actions.itemsLoaded(1, 2, 'search', true)).toMatchSnapshot();
    });

    it('itemsLoadingError should create LOAD_ITEMS_ERROR action', () => {
      expect(actions.itemsLoadingError(1, 2, 'search', 'error')).toMatchSnapshot();
    });

    it('itemLoaded should create LOAD_ITEM_SUCCESS action', () => {
      expect(actions.itemLoaded(1, 2, 'search', 'item')).toMatchSnapshot();
    });

    it('itemLoadingError should create LOAD_ITEM_ERROR action', () => {
      expect(actions.itemLoadingError(1, 2, 'search', 'item', 'error')).toMatchSnapshot();
    });
  });
});
