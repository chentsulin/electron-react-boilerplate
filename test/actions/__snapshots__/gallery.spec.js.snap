// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions gallery itemLoaded should create LOAD_ITEM_SUCCESS action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": "item",
  "type": "redux/Gallery/LOAD_ITEM_SUCCESS",
}
`;

exports[`actions gallery itemLoadingError should create LOAD_ITEM_ERROR action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": Object {
    "error": "error",
    "item": "item",
  },
  "type": "redux/Gallery/LOAD_ITEM_ERROR",
}
`;

exports[`actions gallery itemsLoaded should create LOAD_ITEMS_SUCCESS action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": Object {
    "hasNextPage": true,
  },
  "type": "redux/Gallery/LOAD_ITEMS_SUCCESS",
}
`;

exports[`actions gallery itemsLoadingError should create LOAD_ITEMS_ERROR action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": "error",
  "type": "redux/Gallery/LOAD_ITEMS_ERROR",
}
`;

exports[`actions gallery loadItems should create LOAD_ITEMS action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "type": "redux/Gallery/LOAD_ITEMS",
}
`;

exports[`actions gallery markItem should create MARK_ITEM action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": Object {
    "id": 3,
    "marked": true,
  },
  "type": "redux/Gallery/MARK_ITEM",
}
`;

exports[`actions gallery saveScrollPosition should create SAVE_SCROLL_POSITION action 1`] = `
Object {
  "meta": Object {
    "galleryId": 2,
    "moduleId": 1,
    "searchQuery": "search",
  },
  "payload": Object {
    "oldWidth": 4,
    "scrollPosition": 3,
  },
  "type": "redux/Gallery/SAVE_SCROLL_POSITION",
}
`;
